src/logic/signupLogic.ts

```diff
@@ -0,0 +1,118 @@
+
+export enum SignupModalView {
+    EMAIL_PROMPT = 'EMAIL_PROMPT',
+    DEPLOYMENT_OPTIONS = 'DEPLOYMENT_OPTIONS',
+}
+
+export enum Realm {
+    hosted = 'hosted',
+    cloud = 'cloud',
+}
+
+type HubSpotContact = {
+    id: string // Hubspot contact ID (number as string)
+    properties: Record<string, string>
+    created_at: string // ISO 8601
+    updated_at: string // ISO 8601
+    archived: boolean
+    archived_at: string | null // ISO 8601
+}
+
+export type HubSpotContactResponse = {
+    success: boolean
+    result: string | HubSpotContact
+}
+
+async function createContact(email: string) {
+    const url = process.env.GATSBY_POSTHOG_API_HOST + '/create_web_contact'
+    const body = new FormData()
+    body.append('email', email)
+    return (
+        await fetch(url, {
+            method: 'POST',
+            body,
+        })
+    ).json() as Promise<HubSpotContactResponse>
+}
+
+async function updateContact(email: string, properties: Record<string, string>) {
+    const url = process.env.GATSBY_POSTHOG_API_HOST + '/update_web_contact'
+    const body = new FormData()
+    body.append('email', email)
+    for (const [property, value] of Object.entries(properties)) {
+        body.append(property, value)
+    }
+    return (
+        await fetch(url, {
+            method: 'POST',
+            body,
+        })
+    ).json() as Promise<HubSpotContactResponse>
+}
+
+export const signupLogic = kea({
+    actions: {
+        setModalView: (view: SignupModalView) => ({ view }),
+        setEmail: (email: string) => ({ email }),
+        submitForm: true,
+        skipEmailEntry: true,
+        reportModalShown: true,
+        reportDeploymentTypeSelected: (deploymentType: Realm, nextHref?: string) => ({ deploymentType, nextHref }),
+    },
+    reducers: {
+        modalView: [
+            SignupModalView.EMAIL_PROMPT as SignupModalView,
+            {
+                setModalView: (_, { view }: { view: SignupModalView }) => view,
+            },
+        ],
+        email: [
+            '',
+            {
+                setEmail: (_, { email }: { email: string }) => email,
+            },
+        ],
+    },
+    connect: {
+        values: [posthogAnalyticsLogic, ['posthog']],
+    },
+    listeners: ({ actions, values }) => ({
+        submitForm: async () => {
+            const { posthog, email } = values
+            if (email && isValidEmailAddress(email)) {
+                try {
+                    posthog?.identify(email, { email }) // use email as distinct ID; also set it as property
+                    posthog?.capture('signup: submit email')
+                    await createContact(email)
+                    actions.setModalView(SignupModalView.DEPLOYMENT_OPTIONS)
+                } catch (err) {
+                    posthog?.capture('signup: failed to create contact', { message: err })
+                }
+            }
+        },
+        skipEmailEntry: async () => {
```

Comment:
Nit: might be good to call this reportSkipEmailEntry so it fits in w/ the rest

Reply:
Because this calls `setModalView`, it's an action ("skip") and a side effect ("report") in one. Think I'll keep the name.